File: webapp/models/submission.py
#!/usr/bin/env python
# -*- coding: iso-8859-15 -*-
import os
import sys
from sqlalchemy import Column, ForeignKey, Integer, String, DECIMAL, Float, VARBINARY, DateTime
from sqlalchemy.ext.declarative import declarative_base
from sqlalchemy.orm import relationship
from sqlalchemy import create_engine
from sqlalchemy.sql import func
from sqlalchemy.dialects.mysql import YEAR

Base = declarative_base()



class SubmissionTable(Base):
    __tablename__ = "SubmissionTable"
    ID = Column(String(36), nullable=False, primary_key=True)
    SubmissionDate = Column(DateTime(timezone=True), server_default=func.now())
    User = Column(String(100), nullable=True)
    AnalysisType = Column(String(25), nullable=True)
    IPaddress= Column(String(16), nullable=True)

class isolateSubmission(Base):
    __tablename__ = "isolateSubmission"
    ID = Column(String(36), nullable=False, primary_key=True)
    submissionType = Column(String(10), nullable=False)
    SubmissionDate = Column(DateTime(timezone=True), server_default=func.now())
    isolateName = Column(String(100), nullable=True)
    isolateSource = Column(String(100), nullable=True)
    isolateHost = Column(String(100), nullable=True)
    placeOfIsolation = Column(String(100), nullable=True)
    yearOfIsolation = Column(YEAR(4))
    country = Column(String(100), nullable=True)
    latitude = Column(DECIMAL(10,8))
    longitude = Column(DECIMAL(11,8))
    pubmedID = Column(String(50), nullable=True)
    genomeAccession = Column(String(50), nullable=True)
    email = Column(String(255), nullable=True)
    submitterName = Column(String(255), nullable=True)


class mlvaSubmission(Base):
    __tablename__ = "mlvaSubmission"
    ID = Column(String(36), nullable=False, primary_key=True)
    ms01 = Column(Float(10,2), nullable=True)
    ms03 = Column(Float(10,2), nullable=True)
    ms20 = Column(Float(10,2), nullable=True)
    ms21 = Column(Float(10,2), nullable=True)
    ms22 = Column(Float(10,2), nullable=True)
    ms23 = Column(Float(10,2), nullable=True)
    ms24 = Column(Float(10,2), nullable=True)
    ms26 = Column(Float(10,2), nullable=True)
    ms27 = Column(Float(10,2), nullable=True)
    ms28 = Column(Float(10,2), nullable=True)
    ms30 = Column(Float(10,2), nullable=True)
    ms31 = Column(Float(10,2), nullable=True)
    ms33 = Column(Float(10,2), nullable=True)
    ms34 = Column(Float(10,2), nullable=True)
    isolateID = Column(String(36), ForeignKey("isolateSubmission.ID"), nullable=False)


class mstSubmission(Base):
    __tablename__ = "mstSubmission"
    ID = Column(String(36), nullable=False, primary_key=True)
    cox18 = Column(Integer(), nullable=True)
    cox2 = Column(Integer(), nullable=True)
    cox20 = Column(Integer(), nullable=True)
    cox22 = Column(Integer(), nullable=True)
    cox37 = Column(Integer(), nullable=True)
    cox5 = Column(Integer(), nullable=True)
    cox51 = Column(Integer(), nullable=True)
    cox56 = Column(Integer(), nullable=True)
    cox57 = Column(Integer(), nullable=True)
    cox61 = Column(Integer(), nullable=True)
    isolateID = Column(String(36), ForeignKey("isolateSubmission.ID"), nullable=False)


engine = create_engine('mysql+pymysql://burnetii:crazyburnetii@localhost/MLVA')
Base.metadata.create_all(engine)

File: webapp/static/js/submission.js
$(document).ready(function()
{

dict = {'mlva' : '<table class="query_table"><tr><th class="query_table_header">ms01</th><th class="query_table_header">ms03</th><th class="query_table_header">ms20</th><th class="query_table_header">ms21</th><th class="query_table_header">ms22</th><th class="query_table_header">ms23</th><th class="query_table_header">ms24</th><th class="query_table_header">ms26</th><th class="query_table_header">ms27</th><th class="query_table_header">ms28</th><th class="query_table_header">ms30</th><th class="query_table_header">ms31</th><th class="query_table_header">ms33</th><th class="query_table_header">ms34</th></tr><tr><td><input class="tr_entry" name="ms01" type="number" min="0" step="any"></td><td><input class="tr_entry" name="ms03" type="number" min="0" step="any"></td><td><input class="tr_entry" name="ms20" type="number" min="0" step="any"></td><td><input class="tr_entry" name="ms21" type="number" min="0" step="any"></td><td><input class="tr_entry" name="ms22" type="number" min="0" step="any"></td><td><input class="tr_entry" name="ms23" type="number" min="0" step="any"></td><td><input class="tr_entry" name="ms24" type="number" min="0" step="any"></td><td><input class="tr_entry" name="ms26" type="number" min="0" step="any"></td><td><input class="tr_entry" name="ms27" type="number" min="0" step="any"></td><td><input class="tr_entry" name="ms28" type="number" min="0" step="any"></td><td><input class="tr_entry" name="ms30" type="number" min="0" step="any"></td><td><input class="tr_entry" name="ms31" type="number" min="0" step="any"></td><td><input class="tr_entry" name="ms33" type="number" min="0" step="any"></td><td><input class="tr_entry" name="ms34" type="number" min="0" step="any"></td></tr></table>',
	'mst' : '<table class="query_table"><tr> <th class="query_table_header">COX2</th><th class="query_table_header">COX5</th><th class="query_table_header">COX18</th><th class="query_table_header">COX20</th><th class="query_table_header">COX22</th><th class="query_table_header">COX37</th><th class="query_table_header">COX51</th><th class="query_table_header">COX56</th><th class="query_table_header">COX57</th><th class="query_table_header">COX61</th></tr><tr><td><input class="tr_entry" name="cox2" type="number"></td><td><input class="tr_entry" name="cox5" type="number"></td><td><input class="tr_entry" name="cox18" type="number"></td><td><input class="tr_entry" name="cox20" type="number"></td><td><input class="tr_entry" name="cox22" type="number"></td><td><input class="tr_entry" name="cox37" type="number"></td><td><input class="tr_entry" name="cox51" type="number"></td><td><input class="tr_entry" name="cox56" type="number"></td><td><input class="tr_entry" name="cox57" type="number"></td><td><input class="tr_entry" name="cox61" type="number"></td></tr></table>'}
$('.js-example-basic-single').select2();

$('#select_panel').change(function() {
var key = ($("#select_panel option:selected").val());
var panel = dict[key]
$('#table_div').html(panel)
})

window.validateform = function(){
	if ($("#aof").prop('checked') == false){
	throw_acknowledgment_error()
		return false;
	}
	
};

function throw_acknowledgment_error(){
	$('#agreement').html(" !!! Please agree to fair use by ticking this checkbox")
}

})


